### DISCUSSION TITLE
Read only files?

### DISCUSSION THREAD

**Discussion by baschwar** [dz0ny/klipper-prusa-mk3s#42] [Q&A]
I'm new to Klipper and have used a template for the Prusa Mk3 to get things set up. 

I'm trying to copy the macros.cfg file and make a writeable copy but even connecting to the filesystem and doing chmod 644 or 755 will not allow the file to be writeable within the Klipper UI. It still stays read-only. 

I was able to make a duplicate of the printer-template.cfg in the UI and it is writeable. But, I'm unable to duplicate macros.cfg in the UI. 
What am I missing?

**Comment by confuzled-dev** [comment]
What I did was call an additional file called custom_macros.cfg and do all
your custom stuff there.

I did the same with the eisny cfg for the same reason

**Reply by baschwar** [reply]
I tried making a new file in the terminal and it is still read only. Even though permissions are wide open 755

**Comment by confuzled-dev** [comment]
Put it in the same folder as your printer.cfg

Just create a new folder there and add all your extra macros to it
